### PR TITLE
Friendship milestone

### PR DESCRIPTION
**Pull request by jstloyal**
- Create a Friendship model with the association with the User model.
- Create friendship method within the User model

### CODE REVIEW

**Comment by uryelah** [comment]
Hi ** Loyalty Adetayo**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good start to this milestone. Now it does not end in just the models, make sure to read the requirements again and then check the comments below to see what you are missing. Good coding! 💪 

# Required Changes 🏗️ 🐛 💡

- [ ] Make sure all your linter checks are passing.

- [ ] Edit the timeline_posts method so it gets posts from the user AND all of the user friends.

- [ ] There should be buttons in the user index and show page to request friendship from a user.

- [ ] You'll need a friendship controller so the user can:

  -invite friends

  - see friendships requests 

  - accept friendship request to start a friendship

  - reject a friendship request

-----

 **Kindly fix the pointed out issues before requesting the next code review**

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -3,6 +3,12 @@
     <% @users.each do |user| %>
       <li>
         Name: <%= user.name %>
+        <% if @user_requests.include?(user) %>
+          <button class="request-btn"><%= link_to 'Accept request', friendships_path(friend_id: user.id) , method: :post %></button>
+          <button class="reject-request"><%= link_to 'Reject request', friendships_path(user_id: user.id) , method: :delete, class: "request" %></button>
+        <% else %>
+          <button class="request-btn"><%= link_to 'Friend request', friendships_path(friend_id: user.id), method: :post, class: "request" %></button>
+        <% end %>
```
Comment (maelfosso) [inline app/views/users/index.html.erb:11]: Such logic, if/else statement, must be handled into **view helpers**

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -16,7 +16,7 @@
         </div>
         <div class="menu-item">
           <%= menu_link_to 'All users', users_path %>
-        </div>
+        </div> 
       </div>
       <% if current_user %>
         <%= link_to 'Sign out', destroy_user_session_path, method: :delete %>
```
Comment (maelfosso) [inline app/views/layouts/application.html.erb:22]: Handle this with view helpers

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -20,7 +20,7 @@ def create
   private
 
   def timeline_posts
-    @timeline_posts ||= Post.all.ordered_by_most_recent.includes(:user)
+    @timeline_posts ||= Post.where(user: current_user.friends + [current_user.id]).ordered_by_most_recent
```
Comment (maelfosso) [inline app/controllers/posts_controller.rb:23]: You should move it in the `User` model
```rb
# User model
def friends_and_own_posts
  Post.where(user: (self.friends + self))
end
```

and call it like this
```rb
def timeline_posts
    @timeline_posts ||= current_user.friend_and_own_posts
end
```

**Pull request closed without merging**